wrappers/nodejs/index.js

```diff
@@ -2280,36 +2297,57 @@ class FrameSet {
     }
   }
 
-  __internalGetFrame(stream) {
-    return Frame._internalCreateFrame(this.cxxFrameSet.getFrame(stream));
+  __internalGetFrame(stream, streamIndex) {
+    return Frame._internalCreateFrame(this.cxxFrameSet.getFrame(stream, streamIndex));
+  }
+
+  __internalFindFrameInCache(stream, streamIndex) {
+    for (let i = 0; i < this.cacheMetadata.length; i++) {
+      if (this.cacheMetadata[i].stream !== stream) {
+        continue;
+      }
+      if (!streamIndex || (streamIndex && streamIndex === this.cacheMetadata[i].streamIndex)) {
+        return i;
+      }
+    }
+    return undefined;
   }
 
-  __internalGetFrameCache(stream, callback) {
-    if (! this.cache[stream]) {
-      this.cache[stream] = callback(stream);
+  __internalGetFrameCache(stream, streamIndex, callback) {
+    let idx = this.__internalFindFrameInCache(stream, streamIndex);
+    if (idx === undefined) {
+      this.cache.push(callback(stream, streamIndex));
+      this.cacheMetadata.push({stream: stream, streamIndex: streamIndex});
+      idx = this.cache.length - 1;
     } else {
-      let frame = this.cache[stream];
+      let frame = this.cache[idx];
       if (!frame.cxxFrame) {
         frame.cxxFrame = new RS2.RSFrame();
       }
-      if (! this.cxxFrameSet.replaceFrame(stream, frame.cxxFrame)) {
-        this.cache[stream] = undefined;
+      if (! this.cxxFrameSet.replaceFrame(stream, streamIndex, frame.cxxFrame)) {
+        this.cache[idx] = undefined;
+        this.cacheMetadata[idx] = undefined;
       }
     }
-    return this.cache[stream];
+    return this.cache[idx];
   }
 
   /**
    * Get the frame with specified stream
    *
    * @param {Integer|String} stream stream type of the frame
+   * @param {Integer} streamIndex index of the stream, 0 means the first
+   * matching stream
    * @return {DepthFrame|VideoFrame|Frame|undefined}
    */
-  getFrame(stream) {
+  getFrame(stream, streamIndex = 0) {
     const funcName = 'FrameSet.getFrame()';
-    checkArgumentLength(1, 1, arguments.length, funcName);
+    checkArgumentLength(1, 2, arguments.length, funcName);
     const s = checkArgumentType(arguments, constants.stream, 0, funcName);
-    return this.__internalGetFrameCache(s, this.__internalGetFrame.bind(this));
+    if (arguments.length === 2) {
```

Comment:
duplicate checking above？

Reply:
there can be 1 or 2 arguments, we can only check the 2nd argument type if it's actually provided. So a condition is needed.